### PR TITLE
Truncate tables and rename documents folder on reset

### PR DESCRIPTION
**Pull request by mejo-**
### 📝 Summary

* Resolves: #5904

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits

### CODE REVIEW

**Review by juliushaertl — APPROVED** [review]
👍 For the truncate part, would appreciate a review for my first commit then

File: lib/Command/ResetDocument.php [hunk 1]
```diff
@@ -61,9 +60,7 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 		}
 
 		if ($all) {
-			$fileIds = array_map(static function (Document $document) {
-				return $document->getId();
-			}, $this->documentService->getAll());
+			$fileIds = $this->documentService->getAll();
```
Comment (mejo-) [inline lib/Command/ResetDocument.php:69]: @juliushaertl I don't understand this change. `DocumentService->getAll()` returns an array of `Document` objects, not the file IDs. How is this supposed to work?

Comment (mejo-) [reply lib/Command/ResetDocument.php]: Tested and indeed it broke with your current approach. The old approach didn't work either as `array_map()` cannot be run against a `Generator`. I pushed a fix:

```php
$fileIds = [];
$documents = $this->documentService->getAll();
foreach ($documents as $document) {
	$fileIds[] = $document->getId();
}
```

**Comment by mejo-** [comment]
> would appreciate a review for my first commit then

Apart from the commented part, your changes look good to me @juliushaertl

Could you review my changes another time, given that I implemented the logic to rename the directory and clean up old directories after your last review?

File: lib/Db/DocumentMapper.php [hunk 2]
```diff
@@ -38,12 +39,33 @@ public function find(int $documentId): Document {
...
+			->from($this->getTableName());
+		$result = $qb->executeQuery();
+		$count = (int)$result->fetchOne();
+		$result->closeCursor();
+		return $count;
+	}
+
+	public function clearAll(): void {
+		$qb = $this->db->getQueryBuilder();
+		$qb->delete($this->getTableName())
```
Comment (blizzz) [inline lib/Db/DocumentMapper.php:68]: truncate and delete are different things. Truncate might be faster than delete, but not all DB engines support it afaik. Not sure what is the current state in doctrine. OK for now.

File: lib/Db/SessionMapper.php [hunk 3]
```diff
@@ -137,6 +148,12 @@ public function deleteByDocumentId(int $documentId): int {
 		return $qb->executeStatement();
 	}
 
+	public function clearAll(): void {
+		$qb = $this->db->getQueryBuilder();
+		$qb->delete($this->getTableName())
```
Comment (blizzz) [inline lib/Db/SessionMapper.php:153]: as above

File: lib/Db/StepMapper.php [hunk 4]
```diff
@@ -62,6 +62,12 @@ public function deleteAll(int $documentId): void {
 			->executeStatement();
 	}
 
+	public function clearAll(): void {
+		$qb = $this->db->getQueryBuilder();
+		$qb->delete($this->getTableName())
```
Comment (blizzz) [inline lib/Db/StepMapper.php:67]: as above

**Review by blizzz — APPROVED** [review]

File: lib/Db/DocumentMapper.php (hunk 2 above)
Comment (mejo-) [reply lib/Db/DocumentMapper.php:68]: I tested with 120.000 entries and it was super fast. So I guess it's ok for now :grimacing:

Comment (blizzz) [reply lib/Db/DocumentMapper.php:68]: https://github.com/nextcloud/server/blob/master/apps/user_ldap/lib/Migration/Version1130Date20211102154716.php#L52-L54

**Comment by blizzz** [comment]
Runner 9 was green before, Runner 1 is a known Flaky test → Merging.

**Pull request merged**